### DISCUSSION TITLE
from a closed LineSet to a 2d mesh

### DISCUSSION THREAD

**Discussion by zonderr** [isl-org/Open3D#6327] [General]
Is there a way to go from a closed LineSet to a 2d mesh?
update : the 2D mesh being the surface inside the closed LineSet.

**Comment by benjaminum** [comment]
There is unfortunately no direct function for triangulating polygons. LineSet has some functions for extrusion but not sure if that can help in your case.

From the documentation for LineSet.extrude_rotation
```
import open3d as o3d

line = o3d.t.geometry.LineSet([[0.7,0,0],[1,0,0]], [[0,1]])
spring = line.extrude_rotation(3*360, [0,1,0], resolution=3*16, translation=2)
o3d.visualization.draw([{'name': 'spring', 'geometry': spring}])
```
